### PR TITLE
Fixing jsonparse task

### PR DESCRIPTION
**Pull request by pinebit**
Closes https://app.shortcut.com/chainlinklabs/story/56225/node-get-uint256-failing-during-ethabiencode-type-json-number-cannot-be-converted-to-decimal-decimal
Fixes https://github.com/smartcontractkit/chainlink/issues/7684

### CODE REVIEW

File: core/services/pipeline/task.jsonparse_test.go [hunk 1]
```diff
@@ -192,7 +191,7 @@ func TestJSONParseTask(t *testing.T) {
 			"false",
 			pipeline.NewVarsFrom(nil),
 			[]pipeline.Result{{Value: `{"data": [[0, 1]]}`}},
-			[]interface{}{json.Number("0"), json.Number("1")},
+			[]interface{}{int64(0), int64(1)},
```
Comment (samsondav) [inline core/services/pipeline/task.jsonparse_test.go:194]: Do we perhaps also want to keep the previous case?

File: core/services/pipeline/task.jsonparse.go [hunk 2]
```diff
@@ -110,12 +109,9 @@ func (t *JSONParseTask) Run(_ context.Context, l logger.Logger, vars Vars, input
 		}
 	}
 
-	switch val := decoded.(type) {
-	case json.Number:
-		decoded, err = getJsonNumberValue(val)
-		if err != nil {
-			return Result{Error: multierr.Combine(ErrBadInput, err)}, runInfo
-		}
+	decoded, err = reinterpetJsonNumbers(decoded)
```
Comment (samsondav) [inline core/services/pipeline/task.jsonparse.go:112]: Huh, so this was already defined but we weren't using it?

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/services/pipeline/task.jsonparse.go (hunk 2 above)
Comment (pinebit) [reply core/services/pipeline/task.jsonparse.go:112]: `reinterpetJsonNumbers` was a new function I introduced recently, just did not update this place.

File: core/services/pipeline/task.jsonparse_test.go (hunk 1 above)
Comment (pinebit) [reply core/services/pipeline/task.jsonparse_test.go:194]: After json unmarshalling and processing all `json.Number`s there should be no `json.Number`s left in a tree. We should always convert those to numbers, or fail.

File: core/services/pipeline/task.jsonparse.go (hunk 2 above)
Comment (samsondav) [reply core/services/pipeline/task.jsonparse.go:112]: I cannot see it in this PR

**Review by samsondav — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

File: core/services/pipeline/task.jsonparse.go (hunk 2 above)
Comment (pinebit) [reply core/services/pipeline/task.jsonparse.go:112]: https://github.com/smartcontractkit/chainlink/pull/7434/files#diff-58294c479ec0dfd4ae41124d476e13a9d93777b46cd32c5f46d47eaaba442d0e

**Pull request merged**